src/python/test/test_dxclient.py

```diff
@@ -6424,6 +6424,36 @@ def test_build_app_with_resources(self):
             container_content = dxpy.api.container_list_folder(app_container, {"folder": "/"})
             self.assertIn(file_id, [item["id"] for item in container_content["objects"]])
 
+    @unittest.skipUnless(testutil.TEST_ISOLATED_ENV,
+                         'skipping test that would create apps')
+    def test_build_cluster_app_bootstrap_script_inlined(self):
+        app_name = "cluster_app"
+        cluster_spec = {"type": "spark",
+                        "version": "2.2.0",
+                        "initialInstanceCount": 5,
+                        "bootstrapScript": "clusterBootstrap.py"}
+        app_spec = dict(self.base_app_spec, name=app_name,
+                        regionalOptions = {
+                            "aws:us-east-1": {
+                                "systemRequirements": {
+                                    "cluster_1": {
+                                        "instanceType": "mem2_hdd2_x1",
+                                        "clusterSpec": cluster_spec
+                                    },
+                                    "cluster_2": {
+                                        "instanceType": "mem2_hdd2_x4",
+                                        "clusterSpec": cluster_spec
+                                    }
+                                }
+                            }})
```

Comment:
I would rename this test to `test_build_and_get_cluster_app_bootstrap_script_inlined` and additionally check if subsequent getting of the app (with `dx get app-xxxx`) unpacks the app configuration into `dxapp.json` so that we can later build the app again with `dx build`, e.g.:

```
+                            "aws:us-east-1": {
+                                "systemRequirements": {
+                                    "cluster_1": {
+                                        "instanceType": "mem2_hdd2_x1",
+                                        "clusterSpec": {"type": "spark",
+                                                                    "version": "2.2.0",
+                                                                    "initialInstanceCount": 5,
+                                                                    "bootstrapScript": "clusterBootstrap.py"}
+                                                                    }
+                                    "cluster_2": {
+                                        "instanceType": "mem2_hdd2_x4",
+                                        "clusterSpec": <as above>
+                                    }
+                                }
+                            }}
```

The code for unpaking the app with `dx get` is here: https://github.com/dnanexus/dx-toolkit/blob/master/src/python/dxpy/utils/executable_unbuilder.py#L238

Getting needs to result in dxapp.json whose specification is here: https://wiki.dnanexus.com/dxapp.json

Also is the file `clusterBootstrap.py` region-specific or the same for all regions?